src/test/java/com/hotels/beeju/HiveServer2JUnitRuleTest.java

```diff
@@ -0,0 +1,276 @@
+    String tableName = "my_test_table";
+
+    try (Connection connection = DriverManager.getConnection(server.connectionURL());
+        Statement statement = connection.createStatement()) {
+      String createHql = new StringBuilder(256)
+          .append("CREATE TABLE `" + DATABASE + "." + tableName + "`(`id` int, `name` string) ")
+          .append("ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe' ")
+          .append("STORED AS INPUTFORMAT 'org.apache.hadoop.mapred.TextInputFormat' ")
+          .append("OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'")
+          .toString();
+      statement.execute(createHql);
+    }
+
+    HiveMetaStoreClient client = server.newClient();
+    Table table = client.getTable(DATABASE, tableName);
+    client.close();
+    assertThat(table.getDbName(), is(DATABASE));
+    assertThat(table.getTableName(), is(tableName));
+    assertThat(table.getSd().getCols(),
+        is(Arrays.asList(new FieldSchema("id", "int", null), new FieldSchema("name", "string", null))));
+    assertThat(table.getSd().getInputFormat(), is("org.apache.hadoop.mapred.TextInputFormat"));
+    assertThat(table.getSd().getOutputFormat(), is("org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat"));
+    assertThat(table.getSd().getSerdeInfo().getSerializationLib(),
+        is("org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe"));
+  }
+
+  @Test
+  public void dropTable() throws Exception {
+    String tableName = "my_table";
+    HiveMetaStoreClient client = server.newClient();
+
+    try {
+      Table table = new Table();
+      table.setDbName(DATABASE);
+      table.setTableName(tableName);
+      table.setSd(new StorageDescriptor());
+      table.getSd().setCols(Arrays.asList(new FieldSchema("id", "int", null), new FieldSchema("name", "string", null)));
+      table.getSd().setInputFormat("org.apache.hadoop.mapred.TextInputFormat");
+      table.getSd().setOutputFormat("org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat");
+      table.getSd().setSerdeInfo(new SerDeInfo());
+      table.getSd().getSerdeInfo().setSerializationLib("org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe");
+      client.createTable(table);
+
+      try (Connection connection = DriverManager.getConnection(server.connectionURL());
+          Statement statement = connection.createStatement()) {
+        String dropHql = String.format("DROP TABLE %s.%s", DATABASE, tableName);
+        statement.execute(dropHql);
+      }
+
+      try {
+        client.getTable(DATABASE, tableName);
+        fail(String.format("Table %s.%s was not deleted", DATABASE, tableName));
+      } catch (NoSuchObjectException e) {
+        // expected
+      }
+    } finally {
+      client.close();
+    }
+  }
+
+  @Test
+  public void addPartition() throws Exception {
+    String tableName = "my_table";
+    HiveMetaStoreClient client = server.newClient();
+
+    try {
+      Table table = new Table();
```

Comment:
Looks like we do this a couple of times can we remove some of the duplication to a private method?